packages/frontend/src/pages/Tile/components/TableBanner/TransferOwnership.tsx

```diff
@@ -0,0 +1,70 @@
+import { useMutation } from '@apollo/client'
+import { Flex, Text, VStack } from '@chakra-ui/react'
+import { Button, useToast } from '@opengovsg/design-system-react'
+
+import { UPSERT_TABLE_COLLABORATOR } from '@/graphql/mutations/tiles/upsert-table-collaborator'
+import { GET_TABLE } from '@/graphql/queries/tiles/get-table'
+
+import { useTableContext } from '../../contexts/TableContext'
+
+import { useShareModalContext } from './ShareModalContext'
+
+const TransferOwnership = () => {
+  const { tableId } = useTableContext()
+  const { emailToTransfer, setEmailToTransfer } = useShareModalContext()
+
+  const toast = useToast({
+    status: 'success',
+    duration: 3000,
+    isClosable: true,
+  })
+
+  const [transferOwnership] = useMutation(UPSERT_TABLE_COLLABORATOR, {
+    refetchQueries: [GET_TABLE],
+    awaitRefetchQueries: false,
+    variables: {
+      input: {
+        tableId,
+        email: emailToTransfer ?? '',
```

Comment:
is the nullish check necessary since the `emailToTransfer` should always be a valid email

Reply:
for typing only